swarm/src/handler.rs

```diff
@@ -125,10 +125,11 @@ pub trait ConnectionHandler: Send + 'static {
 
     /// Returns until when the connection should be kept alive.
     ///
-    /// This method is called by the `Swarm` after each invocation of
-    /// [`ConnectionHandler::poll`] to determine if the connection and the associated
-    /// [`ConnectionHandler`]s should be kept alive as far as this handler is concerned
-    /// and if so, for how long.
+    /// `Swarm` checks if there are still active streams on this connection after
+    /// each invocation of [`ConnectionHandler::poll`]. If no, this method will
+    /// be called by the `Swarm` to determine if the connection and the associated
+    /// [`ConnectionHandler`]s should be kept alive as far as this handler is
+    /// concerned and if so, for how long.
```

Comment:
Great start! I'd like to see something like:

> By default, connections are considered active and thus kept-alive whilst:
> - There are streams currently being upgraded, see [`InboundUpgrade`] and [`OutboundUpgrade`].
> - There are still active streams, i.e. instances of [`Stream`] where the user **did not** call [`Stream::no_keep_alive`].
> - The `ConnectionHandler` returns `Poll::Ready`.
>
> Only once none of these conditions are true do we invoke this function to determine, whether the connection should be kept alive even further.
> Note that for most protocols, this is not necessary as it represents a completely idle connection with no active and no pending streams.
>
> If you'd like to delay the shutdown of idle connections, consider configuring [`SwarmBuilder::idle_connection_timeout`] in your applications.

Based on this, I think we can delete the remaining text below apart from the `**Note**`.

Reply:
fix documentation 412aa75